src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests_AwaitCompletion.vb

```diff
@@ -384,5 +384,251 @@ public class C
+using System.Threading.Tasks;
+
+class C
+{
+    public C Self => this;
+    public Task Field = Task.CompletedTask;
+    public Task Method() => Task.CompletedTask;
+    public Task Property => Task.CompletedTask;
+    public Task this[int i] => Task.CompletedTask;
+    public Func&lt;Task&gt; Function() => () => Task.CompletedTask;
+    public static Task operator +(C left, C right) => Task.CompletedTask;
+    public static explicit operator Task(C c) => Task.CompletedTask;
+}
+
+static class Program
+{
+    static Task StaticField = Task.CompletedTask;
+    static Task StaticProperty => Task.CompletedTask;
+    static Task StaticMethod() => Task.CompletedTask;
+
+    static async Task Main(Task parameter)
+    {
+        var local = Task.CompletedTask;
+        var c = new C();
+
+        <%= expression %>
+
+        Task LocalFunction() => Task.CompletedTask;
+    }
+}
+</Document>)
+                state.SendTypeChars("aw")
+                Await state.AssertSelectedCompletionItem(displayText:="await", isHardSelected:=True)
+
+                state.SendTab()
+                Assert.Equal($"
+using System;
+using System.Threading.Tasks;
+
+class C
+{{
+    public C Self => this;
+    public Task Field = Task.CompletedTask;
+    public Task Method() => Task.CompletedTask;
+    public Task Property => Task.CompletedTask;
+    public Task this[int i] => Task.CompletedTask;
+    public Func<Task> Function() => () => Task.CompletedTask;
+    public static Task operator +(C left, C right) => Task.CompletedTask;
+    public static explicit operator Task(C c) => Task.CompletedTask;
+}}
+
+static class Program
+{{
+    static Task StaticField = Task.CompletedTask;
+    static Task StaticProperty => Task.CompletedTask;
+    static Task StaticMethod() => Task.CompletedTask;
+
+    static async Task Main(Task parameter)
+    {{
+        var local = Task.CompletedTask;
+        var c = new C();
+
+        {committed}
+
+        Task LocalFunction() => Task.CompletedTask;
+    }}
+}}
+", state.GetDocumentText())
+                Await state.AssertLineTextAroundCaret($"        {committed}", "")
+            End Using
+        End Function
+
+        <WpfFact>
+        Public Async Function DotAwaitCompletionAddsAwaitInFrontOfExpressionAndAppendsConfigureAwait() As Task
+            Using state = TestStateFactory.CreateCSharpTestState(
+                <Document><![CDATA[
+using System.Threading.Tasks;
+
+public class C
+{
+    public static async Task Main()
+    {
+        Task.CompletedTask.$$
+    }
+}
+]]>
+                </Document>)
+                state.SendInvokeCompletionList()
+                Await state.AssertCompletionItemsContainAll("await", "awaitF")
+                state.SendTypeChars("a")
+                Await state.AssertSelectedCompletionItem(displayText:="await", isHardSelected:=True)
+                state.SendTypeChars("f")
+                Await state.AssertSelectedCompletionItem(displayText:="awaitF", isHardSelected:=True)
```

Comment:
I think that the trick with `FilterText = "awaitF"` does what we wanted to achieve. I'm still not convinced that it is the right solution, because it breaks the mental model of users, familiar with the behavior of the completion list filter. Users expect the list to behave as if "awaitf" is offered, which is different from what the list does now. If we made the f upper case in the display text ("awaitF") the selecting behavior of the shortcut "af" would not only be more apparent but also more discoverable. Other options that we should consider

* "await false" with FilterText "awaitFalse"
* awaitFalse
* awaitConfigureAwaitFalse
* awaitAndConfigureAwaitFalse
* "await and ConfigureAwait(false)" with FilterText "awaitAndConfigureAwaitFalse"

The last one would allow typing `someTask.aa[tab]`.